src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -26,18 +25,15 @@
 
     private final List<JsonAdaptedSuperGroup> superGroups = new ArrayList<>();
 
-    private final List<JsonAdaptedSubGroup> subGroups = new ArrayList<>();
 
     /**
      * Constructs a {@code JsonSerializableAddressBook} with the given persons.
      */
     @JsonCreator
     public JsonSerializableAddressBook(@JsonProperty("persons") List<JsonAdaptedPerson> persons,
-        @JsonProperty("superGroups") List<JsonAdaptedSuperGroup> superGroups,
-        @JsonProperty("subGroups") List<JsonAdaptedSubGroup> subGroups) {
+        @JsonProperty("superGroups") List<JsonAdaptedSuperGroup> superGroups) {
```

Comment:
Possible issue here but hopefully when parsing superGroups into json you make sure to include subgroup info if not we lose all the data between restarts hahaha